### DISCUSSION TITLE
[Nostr plugin] Cant set custom node config with NWC - 504 Bad Gateway

### DISCUSSION THREAD

**Discussion by sudocarlos** [btcpayserver/btcpayserver#6216] [Plugins & integrations]
I am trying to use Nostr Wallet Connect as the Custom node option for Lightning, but it results in a 504 error after clicking Save or Test. My Lightning node (Alby Hub, tried with 1.6.0, 1.7.2 and 1.8.0) sees the connection request and responds, but I don't know whats happening in BTCPay Server. 

BTCPay Server v1.13.5+30d91d0bd (on Start9 v0.3.5)
Nostr plugin 1.1.13.0

**Comment by G12c4** [comment]
Same here, also using Alby Hub

**Reply by sudocarlos** [reply]
THANK YOU! I thought I was the only one :sweat_smile:

**Reply by G12c4** [reply]
I used Custom Node LNDhub (getalby.com) instead, but I would prefer to use NWC instead because then I can set a budget for that connection.

**Reply by sudocarlos** [reply]
Thank you! Good work around for now as long as I have an alby account.

**Reply by johnheenan** [reply]
The assumption is that NWC can be used to make one touch zap payments using a lightning account with btcpay.

Based on issues I have raised below it looks like the following is needed:

1. NNostr docker fragment needs to be running with btcpay
2. A limited nostr client plugin needs to be written for btcpay the can issue and retrieve private NIP-67 events to the NNostr relay with btcpay. See issue below for what Nip-67 is.
3. The same plugin needs to both request set of new NWC parameters and retrieve existing ones for a store admin through a UI

It is not necessary to keep the limited nostr client btcpay plugin running. However it can be be kept waiting indefintely in a low resource use event loop without polling, becuase that is what a web socker does and btcpay, I think, allows this.

Issues at https://github.com/Kukks/NNostr/issues/28

At the moment we can only speculate. So far there is no known discussion and no complete implementation.

Can the existing Nostr plugin be expanded instead of writing a new one as a nostr client? I guess so. However the private NIP-67 mechanism makes no sense unless there are plans to allow multiple listening nostr client plugins per store. There would need to be limits on what the btcpay nostr client can do. One issue is that there is only one nostr server per btcpay server but there are multiple stores. The nostr client communication would need to be carefully controlled, such as the private key or keys used and the relay used for the client. The secret returned for NWC from NIP-67 for the NIP-47 `nostr+walletconnect://` uri would in fact need to be changed to a secret individual store wide one, with the original secret sent in NIP-67 usable as a store id.
